### DISCUSSION TITLE
Root gone in Magisk After updating to 2023101300 from 2023100300 using both avbroot and avbroot.py

### DISCUSSION THREAD

**Discussion by Tryptamine9** [chenxiaolong/avbroot#192] [Q&A]
Hello. I thought this would be a great place to ask for help, and to potentially report a bug to this excellent project! Ever since I was interested in expanding the capabilities of my mobile device back when I was an iPhone user, I was wary about the kind of security implications it could have. I decided to never root my Android unless I could generate custom keys for the recovery, and a month later found this project! You can't believe my joy since, and it's been an absolute pleasure to use, thank you!

I updated to 2023100300 as usual using your excellent instructions. This update was a bit weird. I was still using avbroot.py as I hadn't looked at the site in a while and seen the new Rust version... When I booted after the update, it took over 10 minutes at the "loading GrapheneOS screen" so I restarted. Then it said "Your phone has no operating system installed and will not boot, Please push a button to power down." You can't believe my horror! Had I somehow made a differently signed image and loaded it? Thankfully not, since after a reboot it came back, with full root and su.

So today after the release of 2023101300 (full Android 14 port) I patched it with the NEW avbroot first, and sideloaded it. All went well, except I had no su after the fact. Magisk shows "N/A" under "Installed" even though I put the Magisk 2.52.apk file in the script.

I'd appreciate some help. I have great backups, don't mind loosing data as I loose it to random factory wipes when I set off my security system anyways. I plan on trying a full wipe and reinstall of GrapheneOS using their ./flash-all script and then your instructions from the begining.

Sorry for being so long-winded, its my nature. Thanks once more for this project and giving your time! I'm compiling GrapheneOS from source and would have it done if it wasn't for Vanadium refusing to build trichrome_chrome_32_64.apk! Have a great day!

**Comment by chenxiaolong** [comment]
Glad you're liking avbroot!

> When I booted after the update, it took over 10 minutes at the "loading GrapheneOS screen" so I restarted. Then it said "Your phone has no operating system installed and will not boot, Please push a button to power down." You can't believe my horror! Had I somehow made a differently signed image and loaded it? Thankfully not, since after a reboot it came back, with full root and su.

That's really strange. If you saw the GrapheneOS logo, then the signed boot image was successfully loaded, but I'm not sure why it would've gotten stuck there. Glad the reboot fixed it though. It sounds like Android's A/B rollback feature worked as intended.

> Magisk shows "N/A" under "Installed" even though I put the Magisk 2.52.apk file in the script.

Could you try installing the Magisk 26.3 APK and also repatching/reflashing GrapheneOS with that version? If I remember correctly, there was a Magisk change somewhere in 25.x or 26.x, where mismatched versions between the Magisk app and the Magisk system components may cause `N/A` to be shown.

If you want to confirm that avbroot patched in the Magisk system components correctly, you can run:

```
adb shell magisk -V
```

That should print out the version of the system components.

> I plan on trying a full wipe and reinstall of GrapheneOS using their ./flash-all script and then your instructions from the begining.

I don't think this will make a difference unless you need/want to downgrade GrapheneOS. Sideloading an OTA from recovery overwrites all of the same partitions that the flash-all script does.

> Sorry for being so long-winded, its my nature.

No worries! More details is always better than less, especially for troubleshooting.

**Reply by chenxiaolong** [reply]
It looks like some folks are encountering the `No valid operating system installed` issue with unmodified GrapheneOS 14:

* https://discuss.grapheneos.org/d/8036-grapheneos-version-2023101100-released/20
* https://www.reddit.com/r/GrapheneOS/comments/175s9hx/grapheneos_version_2023101100_released/k4js0q5/

The GrapheneOS folks seems to think it's a firmware bug and I'm inclined to agree. I haven't yet sent in my old device for a trade-in, so I gave things a try.

With `cheetah-ota_update-2023101300.zip`, I get [PXL_20231016_002137283](https://github.com/chenxiaolong/avbroot/assets/646253/3ec3d243-cc70-4dfb-9b94-d193a9bde862) just by continuously rebooting. I can trigger it a lot more frequently by rebooting into different modes (eg. `adb reboot recovery` or `adb reboot rescue`). Sometimes, just powering off and powering on can get things back into a good state. Other times, I have to go into the bootloader menu and boot into recovery or rescue mode to get things "unstuck".

I can't reproduce the issue with the Pixel stock OS though.

I don't think there's much I can do to work around this specific issue in avbroot. GrapheneOS's vbmeta partition structure isn't 100% identical to the Pixel stock firmware. It might be triggering some obscure bug in the bootloader.

EDIT: This seems to be the most active upstream report of this problem: https://discuss.grapheneos.org/d/8450
